src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Telemetry/ProjectTelemetryEvent.cs

```diff
@@ -12,15 +14,32 @@ public class ProjectTelemetryEvent : TelemetryEvent
             string nuGetVersion,
             string projectId,
             NuGetProjectType nuGetProjectType,
-            bool isPRUpgradable) :
+            bool isPRUpgradable,
+            string fullPath) :
             base(ProjectInformationEventName, new Dictionary<string, object>
                 {
                     { nameof(NuGetProjectType), nuGetProjectType },
                     { nameof(NuGetVersion), nuGetVersion },
-                    { nameof(ProjectId), projectId.ToString() },
-                    { IsPRUpgradable, isPRUpgradable }
+                    { nameof(ProjectId), projectId },
+                    { IsPRUpgradable, isPRUpgradable },
                 })
         {
+            if (nuGetVersion == null)
+            {
+                throw new ArgumentNullException(nameof(nuGetVersion));
+            }
+
+            if (projectId == null)
+            {
+                throw new ArgumentNullException(nameof(projectId));
+            }
+
+            if (fullPath == null)
```

Comment:
Do we need to check if `fullPath` is empty string?

Reply:
That is part of the Code Analysis recommendation listed in GlobalSuppressions. See changes to GlobalSuppressions.cs in this PR